artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingManagerImpl.java

```diff
@@ -196,7 +197,11 @@ public void onChange() {
    private void reapplySettings() {
       for (PagingStore store : stores.values()) {
          AddressSettings settings = this.addressSettingsRepository.getMatch(store.getAddress().toString());
-         store.applySetting(settings);
+         try {
+            store.applySetting(settings);
+         } catch (ActiveMQException e) {
+            throw new RuntimeException(e);
+         }
```

Comment:
One aspect of this change is that if it fails the validation and throws applying for one address, this bit here looping over all the addresses means that it wont even try for the addresses following one that fails. The feels potentially problematic as it could leave them with unexpected settings, and yet its entirely possible _their_ settings would have applied ok (i.e theirs may have changed but yet not be invalid).

Reply:
That's true.